generators/typescript/sdk/generator/src/TemplateGenerator.ts

```diff
@@ -291,7 +295,9 @@ export class TemplateGenerator {
         }
         return FdrSnippetTemplate.Template.generic({
             imports: [],
-            templateString: this.getAsNamedParameterTemplate(name, `{\n${childTabs}${TEMPLATE_SENTINEL}\n${selfTabs}}`),
+            templateString: isUnionType
+                ? this.getAsNamedParameterTemplate(name, `${TEMPLATE_SENTINEL}\n`)
+                : this.getAsNamedParameterTemplate(name, `{\n${childTabs}${TEMPLATE_SENTINEL}\n${selfTabs}}`),
```

Comment:
is it worth adding a comment here to understand why the template string changes if it is a union? Definitely a little hard to grok what's happening here